### DISCUSSION TITLE
r2r commandline changes

### DISCUSSION THREAD

**Discussion by trufae** [radareorg/radare2#20249] [Q&A]
There are some flags of r2r that are "confusing" or may be changed, i propose those to be done for 5.8:

* -L should be for listing plugins, aka types of tests (asm, json, ..)
* -F [dir] -> 
* -L should be implicit when using -V .. but maybe we want r2r -V to print version info and use -r
* ~-r and -m should be removed and use env vars instead~
* -u do not git pull - the description should contain the word "update" or use a different flag
* -f [file] -> file to load when running json tests.. maybe this can be an env var too

```
$ r2r -h
Usage: r2r [-qvVnL] [-j threads] [test file/dir | @test-type]
 -C [dir]     chdir before running r2r (default follows executable symlink + test/new
 -F [dir]     run fuzz tests (open and default analysis) on all files in the given dir
 -L           log mode (better printing for CI, logfiles, etc.)
 -V           verbose
 -f [file]    file to use for json tests (default is bins/elf/crackme0x00b)
 -h           print this help
 -i           interactive mode
 -j [threads] how many threads to use for running tests concurrently (default is 8)
 -n           do nothing (don't run any test, just load/parse them)
 -o [file]    output test run information in JSON format to file
 -q           quiet
 -s [ignore]  Set R2R_SKIP_(xxx)=1 to skip running those tests
 -t [seconds] timeout per test (default is (30*60))
 -u           do not git pull/clone test/bins
 -v           show version

R2R_SKIP_ARCHOS=1  # do not run the arch-os-specific tests
R2R_SKIP_JSON=1    # do not run the JSON tests
R2R_SKIP_FUZZ=1    # do not run the rasm2 tests
R2R_SKIP_UNIT=1    # do not run the rasm2 tests
R2R_SKIP_CMD=1     # do not run the rasm2 tests
R2R_SKIP_ASM=1     # do not run the rasm2 tests

Supported test types: @asm @json @unit @fuzz @arch @cmds
OS/Arch for archos tests: darwin-arm
```

**Comment by alex-bender** [comment]
- Could you add some context to `-F [dir] ->`?
- Why it's better to remove `-r` or\and `-m`?
- +1 for `-u` update
why it's better to use env vars than options in general?

**Comment by trufae** [comment]
@alex-bender sure!

* -F is used to specify the path where the fuzzed bins are, r2r will run r2 -qAA on all of them
* -r is used to print output in r2 commands in the rest of r2 tools. so its confusing, also, i think is better to just use r2 from PATH and let the user control that with environment instead otherwise it can result on running one bin with different r2 libs and create even more confussion (same goes for -m)

imho env is better for things that shouldnt change frequently, also its cleaner to read in the ci scripts

**Reply by trufae** [reply]
i did this https://github.com/radareorg/radare2/pull/20366

**Comment by trufae** [comment]
The -f can be renamed to -J it makes more sense to me

**Comment by trufae** [comment]
-L must list the test types 

and current -L can be an env var too
